src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -317,6 +317,22 @@ public bool IsBaseConstructorInitializer(SyntaxToken token)
         public bool IsQueryExpression(SyntaxNode node)
             => node.Kind() == SyntaxKind.QueryExpression;
 
+        public bool IsQueryKeyword(SyntaxToken token)
+            => token.IsKind(
```

Comment:
💡 This is getting a bit long for an expression-bodied member.

💡 Use a `switch` statement to avoid the array allocation used by `params` argument.